src/shared/settings.ts

```diff
@@ -186,14 +190,11 @@ function createSettingsClass<T extends TypeDescriptor>(section: string, descript
         public get<K extends keyof Inner>(key: K & string, defaultValue?: Inner[K]) {
             try {
                 return this.getOrThrow(key, defaultValue)
-            } catch (error) {
-                this.log(`failed to read key "${key}": ${error}`)
-
-                if (isAutomation() || defaultValue === undefined) {
-                    throw new Error(`Failed to read key "${key}": ${error}`)
+            } catch (e) {
+                if (this.throwInvalid || defaultValue === undefined) {
+                    throw new Error(`Failed to read key "${key}": ${e}`)
```

Comment:
on master branch (not only this PR), this causes 34 messages (each with a big stacktrace) like this:

        Failed to read key "telemetry": TypeError: Failed to cast type
        "object" to convertLegacy: TypeError: Unknown telemetry setting:
        [object Object],[object Object]: Error:
            at TelemetryConfig.get (/Users/runner/work/aws-toolkit-vscode/aws-toolkit-vscode/src/shared/settings.ts:193:27)

For tests like `src/test/shared/settingsConfiguration.test.ts` which intentionally set bad data, those messages are an antifeature. For the rest of the tests we do want that visibility.